girdermedviewer/app/core.py

```diff
@@ -63,99 +62,137 @@ def load_config(self, config_file_path=None):
-            config.get("girder", "api_root")
-        )
-        self.state.default_location = ast.literal_eval(
-            config.get("girder", "default_location", fallback="{}"))
-        self.state.app_name = config.get("ui", "name", fallback="Girder Medical Viewer")
+        self.state.girder_url = self.config.get("girder", "url", fallback=None)
+
+        self.state.app_name = self.config.get("ui", "name", fallback="Girder Medical Viewer")
+
+        self.state.temp_dir = self.config.get("download", "directory", fallback=None)
+        self.state.cache_mode = self.config.get("download", "cache_mode", fallback=None)
 
-        self.state.temp_dir = config.get("download", "directory", fallback=None)
-        self.state.cache_mode = config.get("download", "cache_mode", fallback=None)
+    def connect_girder(self):
+        self.provider = gwc.GirderProvider(
+            value=self.state.api_url,
+            trame_server=self.server
+        )
+        self.ctrl.provider_logout.add(self.provider.logout)
+        self.provider.register_layout(self.ui)
+        self.ui.flush_content()
 
-        self.log_level = config.get("logging", "log_level", fallback="INFO")
+    def disconnect_girder(self):
+        # restore ui to original app ui before setting a new provider as root
+        self.ui = copy(self.app_ui)
+        self.ctrl.provider_logout.remove(self.provider.logout)
+        del self.provider
+        self.ui.flush_content()
 
     def configure_logs(self):
+        log_level = self.config.get("logging", "log_level", fallback="INFO")
         logging.basicConfig(stream=sys.stdout)
         # Silence dependencies logs and keep only the application ones
         logging.getLogger().setLevel(logging.WARNING)
-        logging.getLogger(__package__).setLevel(self.log_level)
+        logging.getLogger(__package__).setLevel(log_level)
+
+    @change("girder_url")
+    def set_girder(self, girder_url, **kwargs):
+        if self.state.girder_connected:
+            self.state.girder_connected = False
+            self.state.default_location = {}
+            self.disconnect_girder()
+
+        if girder_url:
+            api_root = (
+                self.config.get(girder_url, "api_root")
+                if girder_url in self.config
+                else self.config.get("girder", "api_root")
+            )
+            api_url = urljoin(
+                girder_url,
+                api_root
+            )
+            valid_url, self.state.girder_error = is_valid_url(api_url)
+            if valid_url:
+                self.state.api_url = api_url
+                self.connect_girder()
+                self.state.girder_connected = True
+                self.state.default_location = ast.literal_eval(
+                    self.config.get(girder_url, "default_location", fallback="{}")
+                    if girder_url in self.config
+                    else self.config.get("girder", "default_location", fallback="{}")
```

Comment:
this `config.get if else ` could go into a convenient function `get_girder_config(girder_url, "default_location")`